lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+      .listBlobsSegmentedWithPrefixAsync(container, prefix, continuationToken)  
+      .then(result => {
+        const listOfFiles = result.entries;
+        continuationToken = result.continuationToken;
+        let isTruncated = continuationToken ? true : false;         
+        const files = [];     
+        _.each(listOfFiles, file => files.push(_       
+          .chain(file)       
+          .pick('name', 'lastModified')       
+          .set('isTruncated', isTruncated)       
+          .value()          
+        ));      
+        return files;    
+      });
+  }
+
+  remove(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    logger.debug(`Deleting file ${file} in container ${container} `);
+    return this.storage
+      .deleteBlobAsync(container, file)
+      .then(response => logger.info(`Response of blob deletion from cloud storage for ${file} :`, response))
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized('Authorization at the cloud storage provider failed');
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Forbidden, err => {
+        logger.error(err.message);
+        throw new Forbidden('Authentication at the cloud storage provider failed');
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${file}' not found`));
+  }
+
+  download(options) {
+    return utils.streamToPromise(this.storage.createReadStream(
+        options.container, options.remote))
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.remote}' not found`));
+  }
+
+  upload(options, buffer) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        uploadedStream.removeListener('finish', onfinish);
+        uploadedStream.removeListener('error', onerror);
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function onfinish() {
+        cleanup();
+        resolve(JSON.parse(buffer));
+      }
+
+      let uploadedStream = this.storage
+        .createWriteStreamToBlockBlob(
+          options.container, options.remote);
+
+      uploadedStream.on('error', onerror);
+      uploadedStream.on('finish', onfinish);
+      uploadedStream.end(buffer);
+    });
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .upload({
+        container: this.containerName,
+        remote: file
+      }, new Buffer(JSON.stringify(data, null, 2), 'utf8'));
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .download({
+        container: this.containerName,
```

Comment:
this is incorrect. Here instead of `this.containerName` it should be just `containerName` as you can specify in method if you want to read from a diff container

Reply:
done